.github/workflows/e2e.yml

```diff
@@ -0,0 +1,67 @@
+name: End to End Tests
+on: 
+  workflow_dispatch:
+    inputs:
+      automationBranch:
+        description: 'Set the branch to use for automation tests'
+        required: false
+        default: 'develop'
+        type: string
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [macos-latest]
+        vscodeVersion:
+          - stable
+    runs-on: ${{ matrix.os }}
+    steps:
+        - name: Checkout
+          uses: actions/checkout@v2
+          with:
+            path: ./salesforcedx-vscode
+            ref: ${{ github.event.ref }}
+        - uses: actions/setup-node@v3
+          with:
+            node-version-file: 'salesforcedx-vscode/.nvmrc' 
+            cache: npm
+            cache-dependency-path: |
+              salesforcedx-vscode/package-lock.json
+              salesforcedx-vscode-automation-tests/package-lock.json
+        - name: Clone automation tests
+          uses: actions/checkout@v2
+          with:
+            repository: forcedotcom/salesforcedx-vscode-automation-tests
+            path: salesforcedx-vscode-automation-tests
+            ref: ${{ inputs.automationBranch }}
+        - name: Install Extension Dependencies
+          run: |
+            npm install
+          working-directory: salesforcedx-vscode
+        - name: Install Test Dependencies
+          run: |
+            npm install
+          working-directory: salesforcedx-vscode-automation-tests
+        - name: Install the SFDX CLI
+          run: npm install -g sfdx-cli
+        - name: Verify CLI
+          shell: bash
+          run: |
+            set -e
+            sfdx version
+            SFDX_CLI_VERSION=$(sfdx version)
+            if [[ ((`echo $SFDX_CLI_VERSION | grep -c "sfdx-cli/"` > 0))]]
+            then
+              echo "sfdx-cli installed -" $SFDX_CLI_VERSION
+            else
+              echo "The sfdx-cli installation could not be verified"
+              exit 1
+            fi
+        - name: Run headless test
+          uses: GabrielBB/xvfb-action@v1
+          with:
+            run: npm run automation-tests
+            working-directory: salesforcedx-vscode-automation-tests
+          env:
+            VSCODE_VERSION: ${{ matrix.vscodeVersion }}
```

Comment:
What uses `VSCODE_VERSION`?

Reply:
Since we're not using this now, I'll remove it. This is the way that we can access `process.env.VAR` within the automation-tests.

Reply:
Actually, I'm going to go add it to EnvironmentSettigs.ts

Reply:
Update: @jeffb-sfdc is making some changes to factor in the VSCODE_VERSION in the event we would like to use it going forward, so it will stay.